src/main/java/plugily/projects/villagedefense/arena/managers/ShopManager.java

```diff
@@ -101,36 +119,28 @@ public String getDefaultWolfItemName() {
   }
 
   public void openShop(Player player) {
-    Arena arena = ArenaRegistry.getArena(player);
-    if(arena == null) {
-      return;
-    }
-    if(gui == null) {
-      player.sendMessage(plugin.getChatManager().colorMessage(Messages.SHOP_MESSAGES_NO_SHOP_DEFINED));
-      return;
-    }
-    gui.open(player);
+    openMenuConsumer.accept(player);
```

Comment:
Make sure `openMenuConsumer` is not null here in case if some random user gives the `setOpenMenuConsumer` parameter as `null`.